INPTP_AppForFixing/EmployeeDialog.cs

```diff
@@ -1,59 +1,92 @@
-                lbDep.Visible = false;
-                lbDpName.Visible = false;
-                tBDpName.Visible = false;
+            Init(main, employee);
+            bossForEmpl = selectedBoss;
+            if (selectedBoss != null && employee == false)
+            {
+                tBID.Text = selectedBoss.Id.ToString();
+                tBID.Enabled = false;
+                tBFirstName.Text = selectedBoss.FirstName;
+                tBLastName.Text = selectedBoss.LastName;
+                tBJob.Text = selectedBoss.Job;
+                tBSalary.Text = selectedBoss.MonthlySalaryCZK.ToString();
+                dateTimePickerBirthDate.Value = selectedBoss.OurBirthDate;
             }
         }
 
         private void btnOk_Click(object sender, EventArgs e)
         {
-            if (testOfValues())
+            if (validFormInput())
             {
-                main.Bosses.Add(new Boss(Int32.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text)));
-                main.OnBossesChange();
+                if (employee)
+                {
+                    Employee temp = new Employee(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text));
+                    bossForEmpl.InsertEmpl(temp);
+                    main.OnEmplChange();
+                }
+                else
+                {
+                    Boss newBoss = new Boss(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text));
+
+                    // Find boss in hashset by boss ID
+                    Boss bossToEdit = main.Bosses.FirstOrDefault(x => x.Id == newBoss.Id);
+
+                    // If boss doesnt exist, then create it... Else edit existing boss.
+                    if (bossToEdit == null)
+                    {
+                        main.Bosses.Add(newBoss);
+                    }
+                    else
+                    {
+                        bossToEdit = newBoss;
+                        main.Bosses.RemoveWhere(x => x.Id == newBoss.Id);
+                        main.Bosses.Add(bossToEdit);
+                    }
+                    main.OnBossesChange();
+                }
+
                 Close();
             }
-            else {
-                
-                    MessageBox.Show("Bad entry data inserted.", "Error",
-                                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            else
+            {
+                MessageBox.Show("Bad entry data inserted.", "Error",
+                                MessageBoxButtons.OK, MessageBoxIcon.Error);
             }
         }
 
-        private void EmployeeDialog_FormClosed(object sender, FormClosedEventArgs e)
+        private bool validFormInput()
```

Comment:
functions/methods should have verb in name "isValidFormInput","testIfFormInputIsValid",...

Reply:
OK